enterprise/views.py

```diff
@@ -0,0 +1,184 @@
+        """
+        Get the appropriate note for the form.
+        """
+        if required:
+            return _(
+                "*{provider} requires data sharing consent; if consent is not provided, you will"
+                " be redirected to log in page."
+            ).format(provider=provider)
+        else:
+            return _(
+                "*{provider} requests data sharing consent; if consent is not provided, you will"
+                " not be able to get any discounts from {provider}."
+            ).format(provider=provider)
+
+    def get(self, request):
+        """
+        Render a form to collect user input about data sharing consent.
+        """
+        self.patch_mako_lookup()
+
+        # Get the OpenEdX platform name
+        if configuration_helpers is None:
+            raise NotConnectedToEdX(_('Must be installed in an OpenEdX environment to use configuration helpers.'))
+        platform_name = configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME)
+
+        # Get the EnterpriseCustomer for the request; raise an error if there isn't one.
+        customer = get_enterprise_customer_for_request(request)
+        if customer is None:
+            raise Http404
+
+        # Quarantine the user to this module.
+        self.quarantine(request)
+
+        required = customer.enforces_data_sharing_consent(EnterpriseCustomer.AT_LOGIN)
+
+        data = {
+            'platform_name': platform_name,
+            'sso_provider': customer.name,
+            'data_sharing_consent': 'required' if required else 'optional',
+            "messages": {
+                "warning": self.get_warning(customer.name, platform_name, required),
+                "note": self.get_note(customer.name, required),
+            },
+        }
+        if render_to_response is None:
+            raise NotConnectedToEdX(_('Must be installed in an OpenEdX environment to render this response.'))
+        return render_to_response('grant_data_sharing_permissions.html', data, request=request)
+
+    def post(self, request):
+        """
+        Process the above form.
+        """
+        self.lift_quarantine(request)
+        customer = get_enterprise_customer_for_request(request)
+        if customer is None:
+            raise Http404
+        consent_provided = request.POST.get('data_sharing_consent', False)
+        # If the checkbox is unchecked, no value will be sent
+        if get_real_social_auth_object is None:
+            raise NotConnectedToEdX(_('Must be installed in an OpenEdX environment to use this method.'))
+        user = get_real_social_auth_object(request).user
+        ec_user, __ = EnterpriseCustomerUser.objects.get_or_create(
+            user_id=user.id,
+            enterprise_customer=customer,
+        )
+
+        UserDataSharingConsentAudit.objects.update_or_create(
+            user=ec_user,
+            defaults={
+                'state': 'enabled' if consent_provided else 'disabled'
```

Comment:
Use the constants like `UserDataSharingConsentAudit.ENABLED`, not the string values.